app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorFindLessonEndToEnd.kt

```diff
@@ -0,0 +1,367 @@
+          languages = listOf(Language.ENGLISH),
+          subjects = listOf(Subject.AICC),
+          schedule = List(7) { List(12) { 0 } })
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @get:Rule
+  val grantNotificationPermission: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.POST_NOTIFICATIONS)
+
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    val mockFirebaseMessaging = mock<FirebaseMessaging>()
+    whenever(mockFirebaseMessaging.token).thenReturn(Tasks.forResult("mock_fcm_token"))
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getProfiles(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+      onSuccess(listOf(mockStudent)) // Simulate a list of profiles with our beloved Ozymandias
+    }
+    whenever(mockProfileRepository.updateToken(any(), any(), any(), any())).thenAnswer { invocation
+      ->
+      val onSuccess = invocation.arguments[2] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess()
+    }
+    whenever(mockLessonRepository.getLessonsForTutor(eq("mockUid"), any(), any())).thenAnswer {
+        invocation ->
+      val onSuccess = invocation.arguments[1] as (List<Lesson>) -> Unit
+      onSuccess(mockLessons)
+    }
+    whenever(mockLessonRepository.getAllRequestedLessons(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Lesson>) -> Unit
+      onSuccess(mockLessons)
+    }
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    val mockLessonFlow = MutableStateFlow<Lesson?>(mockLessons[0])
+    doReturn(mockLessonFlow).`when`(mockLessonViewModel).selectedLesson
+    mockLessonFlow.value = mockLessons[0]
+
+    networkStatusViewModel =
+        object :
+            NetworkStatusViewModel(
+                application = androidx.test.core.app.ApplicationProvider.getApplicationContext()) {
+          override val isConnected = mockIsConnected
+        }
+  }
+
+  @Test
+  fun tutorProfileCreationAndFirstLessonTest() {
+    var testMapReady = false
```

Comment:
Here testMapReady is not used in the current state still it focus on testing the profile creation and review, it could be removed.